agg-ast/ast/src/serde_test.rs

```diff
@@ -1605,6 +1605,182 @@ mod stage_test {
+                    })
+                },
+            )));
+            let limit_stage: Stage = Limit(20);
+            let search_two_pipeline: Vec<Stage> =
+                vec![Stage::AtlasSearchStage(full_text_search_stage), limit_stage];
+            pipelines.insert("fullTextPipeline".to_string(), search_two_pipeline);
+
+            pipelines
+        }
+
+        // Test #1: Validate that we can parse a basic version of $rankFusion with 0 pipelines
+        test_serde_stage!(
+            rank_fusion_empty_pipelines,
+            expected = Stage::RankFusion(RankFusion {
+                input: RankFusionPipeline {
+                    pipelines: empty_rankfusion_pipeline(),
+                },
+                combination: None,
+                score_details: true
+            }),
+            input = r#"stage: {"$rankFusion": {
+               "input": { "pipelines": { searchOne: []}},
+                "scoreDetails": true,
+            }}"#
+        );
+
+        // Test #2: Validate we can parse $rankFusion stage with a single pipeline
+        test_serde_stage!(
+            rank_fusion_single_pipeline,
+            expected = Stage::RankFusion(RankFusion {
+                input: RankFusionPipeline {
+                    pipelines: single_rankfusion_pipeline(),
+                },
+                combination: None,
+                score_details: false
+            }),
+            input = r#"stage: {"$rankFusion": {
+               "input": { "pipelines": { searchOne: [{ "$vectorSearch" : {"index" : "movie_collection_index", "path" : "title", "queryVector": [10.6, 60.5], "numCandidates": 500} }] } },
+                "scoreDetails": false,
+            }}"#
+        );
+
+        // Test #3: Validate that if a pipeline with the same key appears multiple times, only the last key is respected.
+        test_serde_stage!(
+            rank_fusion_uses_latest_key_to_deduplicate_pipelines,
+            expected = Stage::RankFusion(RankFusion {
+                input: RankFusionPipeline {
+                    pipelines: single_rankfusion_pipeline(),
+                },
+                combination: None,
+                score_details: false
+            }),
+            input = r#"stage: {"$rankFusion": {
+               "input": { "pipelines": {
+               searchOne: [{ "$search": { "index": "hybrid-full-text-search", "phrase": { "query": "star wars", "path": "title"}}}, { "$project": { "title": 1, "released" : 1 } }],
+               searchOne: [{ "$search": { "index": "hybrid-full-text-search", "phrase": { "query": "star wars", "path": "title"}}}, { "$limit": 20 }],
+               searchOne: [{ "$vectorSearch" : {"index" : "movie_collection_index", "path" : "title", "queryVector": [10.6, 60.5], "numCandidates": 500} }] } },
+                "scoreDetails": false,
+            }}"#
+        );
+
+        // Test #4: Validate we parse multiple input pipelines along with specifying weights for each pipeline
```

Comment:
also a tiny nit but I think convention is that the test name should describe the test itself, so these comments are (or should be) redundant